lib/daru/dataframe.rb

```diff
@@ -112,33 +112,17 @@ def from_plaintext path, fields
       # Create DataFrame by specifying rows as an Array of Arrays or Array of
       # Daru::Vector objects.
       def rows source, opts={}
-        first = source.first
-
         raise SizeError, 'All vectors must have same length' \
-          unless source.all? { |v| v.size == first.size }
-
-        index = []
-        opts[:order] ||=
-          case first
-          when Daru::Vector # assume that all are Vectors
-            # FIXME: Not sure, why index is set from vectors
```

Comment:
The `:order` provided by the user should over ride the indexes pointed to by the Vectors. The `||=` in that case should not be there.